### PR TITLE
Musark 220

### PR DESCRIPTION
**Pull request by jarlestabell**

### CODE REVIEW

File: common/src/main/scala/no/uio/musit/microservices/common/extensions/EitherExtensions.scala [hunk 1]
```diff
@@ -53,9 +47,21 @@ object EitherExtensions {
 
   }
 
+  /*Not used
```
Comment (jarlah) [inline common/src/main/scala/no/uio/musit/microservices/common/extensions/EitherExtensions.scala]: remove

File: common/src/main/scala/no/uio/musit/microservices/common/extensions/FutureExtensions.scala [hunk 2]
```diff
@@ -113,6 +113,11 @@ object FutureExtensions {
         for (r <- fr; b <- fb) yield (r += b)
```
Comment (jarlah) [inline common/src/main/scala/no/uio/musit/microservices/common/extensions/FutureExtensions.scala:113]: remove comment. Make a lira issue or sumtin

File: common/src/main/scala/no/uio/musit/microservices/common/utils/ResourceHelper.scala [hunk 3]
```diff
@@ -22,6 +21,18 @@ import no.uio.musit.microservices.common.extensions.FutureExtensions._
...
+          println("Mapping MusitException!")
+          Left(e.toMusitError)
+        }
+      }
+    }
+  }
+
   implicit class EitherExtensionsImp[T](val either: Either[MusitError, T]) extends AnyVal {
 
     /**
```
Comment (jarlah) [inline common/src/main/scala/no/uio/musit/microservices/common/utils/ResourceHelper.scala:26]: remove

File: service_event/src/main/scala/no/uio/musit/microservice/event/dao/EventActorsDao.scala [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    }
+    EventActorsTable ++= relActors
+  }
+
+  def getRelatedActors(eventId: Long): Future[Seq[EventRoleActor]] = {
+    val query = EventActorsTable.filter(evt => evt.eventId === eventId)
+    db.run(query.result)
+  }
+
+  /*
```
Comment (jarlah) [inline service_event/src/main/scala/no/uio/musit/microservice/event/dao/EventActorsDao.scala]: remove

File: service_event/src/main/scala/no/uio/musit/microservice/event/dao/EventActorsDao.scala [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  def insertEventLinkAction(eventLink: EventLink): DBIO[Int] = {
+    insertEventLinkDtoAction(eventLink.toNormalizedEventLinkDto)
+  }
+
+  def insertEventLinkDtoAction(eventLink: EventLinkDto): DBIO[Int] = {
+    //println(s"inserLink: from: ${eventLink.idFrom} relation: ${eventLink.relationId} to: ${eventLink.idTo}")
+    EventLinkTable += eventLink
+  }
+
+  def getRelatedEventDtos(parentId: Long) /*: Future[Seq[(Int, BaseEventDto)]]*/ = {
```
Comment (jarlah) [inline service_event/src/main/scala/no/uio/musit/microservice/event/dao/EventActorsDao.scala]: either specify type or remove comment

File: service_event/src/main/scala/no/uio/musit/microservice/event/dao/EventDao.scala [hunk 6]
```diff
@@ -43,15 +50,26 @@ object EventDao extends HasDatabaseConfig[JdbcProfile] {
...
   private val EventBaseTable = TableQuery[EventBaseTable]
+  private val LinkTable = TableQuery[LinkTable]
 
   def insertBaseAction(eventBaseDto: BaseEventDto): DBIO[Long] =
     EventBaseTable returning EventBaseTable.map(_.id) += eventBaseDto
 
   def selfLink(id: Long) =
     LinkService.local(Some(id), "self", s"/v1/$id")
 
+  /*
```
Comment (jarlah) [inline service_event/src/main/scala/no/uio/musit/microservice/event/dao/EventDao.scala]: remove

File: service_event/src/main/scala/no/uio/musit/microservice/event/dao/EventDao.scala [hunk 7]
```diff
@@ -62,8 +80,12 @@ object EventDao extends HasDatabaseConfig[JdbcProfile] {
 
     val partOfParent = partialEventLink.filter(_ => isPartsRelation).map(_.idFrom)
 
+    val _registeredBy = securityConnection.userName
+
+    val _registeredDate = new java.sql.Timestamp(DateTime.now().getMillis)
+
     val insertBaseAndLinksAction = (for {
-      newEventId <- insertBaseAction(event.baseEventProps.copy(partOf = partOfParent)) //#OLD (EventHelpers.eventDtoToStoreInDatabase(event, partOfParent))
+      newEventId <- insertBaseAction(event.baseEventProps.copy(partOf = partOfParent, registeredBy = Some(_registeredBy), registeredDate = Some(_registeredDate))) //#OLD (EventHelpers.eventDtoToStoreInDatabase(event, partOfParent))
```
Comment (jarlah) [inline service_event/src/main/scala/no/uio/musit/microservice/event/dao/EventDao.scala]: remove comment about old method. Its in git

File: service_event/src/main/scala/no/uio/musit/microservice/event/resource/EventResource.scala [hunk 8]
```diff
@@ -19,22 +19,146 @@
...
+        val maybeEventResult = jsonValidator(request.body.asInstanceOf[JsObject])
+        ResourceHelper.postRootWithMusitResult(EventService.insertAndGetNewEvent(_: Event, true, securityConnection), maybeEventResult, eventToJson)
+    }
+  }
+
+  def postEvent: Action[JsValue] = Action.async(BodyParsers.parse.json) { request: Request[JsValue] =>
+    handlePostEvent(request, JsonEventHelpers.validateEvent)
+  }
+
+  /*#OLD
```
Comment (jarlah) [inline service_event/src/main/scala/no/uio/musit/microservice/event/resource/EventResource.scala]: remove

File: service_event/src/main/scala/no/uio/musit/microservice/event/service/EventService.scala [hunk 9]
```diff
@@ -25,17 +25,45 @@ import no.uio.musit.microservice.event.domain._
...
+
+  private def getEventsFor(eventType: EventType, relation: String, objectUri: String): MusitFuture[Seq[Event]] = {
+    val futEventIds = getEventIdsFor(eventType, relation, objectUri)
+    MusitFuture.traverse(futEventIds)(eventId => getEvent(eventId, true))
+  }
+
+  def getEventsFor(eventType: EventType, relation: String, id: Long): MusitFuture[Seq[Event]] = {
+
+    val objectUri = EventRelations.getObjectUriViaRelation(id, relation).toMusitResult(MusitError(message = s"Unable to get objectUri via relation: $relation"))
+    /*
```
Comment (jarlah) [inline service_event/src/main/scala/no/uio/musit/microservice/event/service/EventService.scala]: remove

**Comment by jarlah** [comment]
all in all, remove all comments that is not documenting the code. Comments that show how things was done before is of no value. Its in git

**Comment by jarlah** [comment]
We can however drop the removal of the comments since @kpmeen is already doing major work on this service.

**Comment by jarlah** [comment]
so therefore it LGTM

**Comment by jarlestabell** [comment]
Did some cleanup, not sure if I got everything, but hopefully did the major ones! :)

**Pull request merged**